### PR TITLE
[SYCL-MLIR][NFC]: Use term 'shared memory' instead of 'local memory'

### PR DESCRIPTION
**Pull request by etiotto**
In `LoopInternalization` use the term 'shared memory' instead of the term 'local memory' (https://www.intel.com/content/www/us/en/docs/oneapi/optimization-guide-gpu/2023-0/shared-local-memory.html#:~:text=The%20shared%20local%20memory%20(SLM,the%20same%20Xe%2Dcore.)

### CODE REVIEW

**Comment by whitneywhtsang** [comment]
But we are promoting accesses to `LocalAccessAddrSpace`, would that mean to `local memory`?

File: polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp [hunk 1]
```diff
@@ -220,110 +220,117 @@ Value getNdItemArgument(FunctionOpInterface func) {
...
 }
 
-/// Create a GlobalOp for workgroup local memory.
-memref::GlobalOp getWorkGroupLocalMemory(gpu::GPUModuleOp module,
-                                         unsigned size) {
+/// Create a GlobalOp for workgroup shared memory.
+memref::GlobalOp getWorkGroupSharedMemory(gpu::GPUModuleOp module,
+                                          unsigned size) {
   assert(size != 0 && "Expecting non-zero size");
-  std::string name("WGLocalMem");
```
Comment (whitneywhtsang) [inline polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp:326]: If possible, I would like to use the same global variable name as LLVM.

File: polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp [hunk 2]
```diff
@@ -1200,29 +1209,30 @@ void LoopInternalization::transform(T loop,
...
-  createLocalBarrier(builder);
+  createWorkGroupBarrier(builder);
   builder.setInsertionPointAfter(loop);
-  createLocalBarrier(builder);
+  createWorkGroupBarrier(builder);
 }
 
-void LoopInternalization::promote(Operation *memref, memref::ViewOp localMemory,
+void LoopInternalization::promote(Operation *memref,
+                                  memref::ViewOp sharedMemory,
```
Comment (whitneywhtsang) [inline polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp:1235]: `localMemory` is renamed to `sharedMemory` here, but renamed to `viewOp` in the declarator.

File: polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp (hunk 1 above)
Comment (etiotto) [reply polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp:326]: Fair request.

File: polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp (hunk 2 above)
Comment (etiotto) [reply polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp:1235]: ty - will change

Comment (etiotto) [reply polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp]: done

**Comment by Naghasan** [comment]
> > But we are promoting accesses to `LocalAccessAddrSpace`, would that mean to `local memory`?
> 
> Shared Local Memory is the correct technical term to use , see: https://www.intel.com/content/www/us/en/docs/oneapi/optimization-guide-gpu/2023-0/shared-local-memory.html#:~:text=The%20shared%20local%20memory%20(SLM,the%20same%20Xe%2Dcore.

Shared memory is the CUDA term, SYCL uses `local` https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_sycl_device_memory_model similar to OpenCL https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html#local-or-local

**Comment by etiotto** [comment]
@whitneywhtsang I have addressed your comments. Any further questions or is this PR good ?

**Review by whitneywhtsang — APPROVED** [review]
> @whitneywhtsang I have addressed your comments. Any further questions or is this PR good ?

I have no further questions. If possible, please don't merge before https://github.com/intel/llvm/pull/9894.

**Review by Naghasan — APPROVED** [review]
I still find odd to not use the term defined by the spec in favor of a term used by an optimization guide, but makes sense if you want to have that reused in other contexts.

**Comment by etiotto** [comment]
> I still find odd to not use the term defined by the spec in favor of a term used by an optimization guide, but makes sense if you want to have that reused in other contexts.

Yeah is always hard to choose the perfect terminology.

**Pull request merged**